completion/api/v1/views.py

```diff
@@ -148,3 +149,61 @@ def post(self, request, *args, **kwargs):  # pylint: disable=unused-argument
             }, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
 
         return Response({"detail": _("ok")}, status=status.HTTP_200_OK)
+
+
+class SubsectionCompletionView(APIView):
+    """
+    Handles API endpoints for the milestones experiments.
+    """
+
+    permission_classes = (permissions.IsAuthenticated, IsUserInUrl)
+
+    def get(self, request, user_id, course_key, subsection_id):
+        """
+        Returns completion for a (user, subsection, course).
+        """
+        def get_completion(all_blocks, block):
```

Comment:
in the services.py, we already do something like this. Additionally, the `is_vertical_complete` method in that file can benefit from your work here. It's optional whether you want to refactor that code.

Reply:
This might be a time for using `utilities.py` - remember that the class in `services.py` exists to be a runtime service accessible to XBlocks.

Reply:
Do you think it makes sense to move this logic into utilities.py and call it from both services.py and views.py?